go-controller/pkg/ovn/secondary_layer3_network_controller.go

```diff
@@ -393,13 +468,41 @@ func (oc *SecondaryLayer3NetworkController) addUpdateNodeEvent(node *kapi.Node,
 		errs = append(errs, errors...)
 	}
 
+	if nSyncs.syncZoneIc && config.OVNKubernetesFeature.EnableInterconnect {
+		if err := oc.zoneICHandler.AddLocalZoneNode(node); err != nil {
+			errs = append(errs, err)
+			oc.syncZoneIcFailed.Store(node.Name, true)
+		} else {
+			oc.syncZoneIcFailed.Delete(node.Name)
+		}
+	}
+
 	err = kerrors.NewAggregate(errs)
 	if err != nil {
 		oc.recordNodeErrorEvent(node, err)
 	}
 	return err
 }
 
+func (oc *SecondaryLayer3NetworkController) addUpdateRemoteNodeEvent(node *kapi.Node, syncZoneIc bool) error {
+	_, present := oc.localZoneNodes.Load(node.Name)
+
+	if present {
+		_ = oc.deleteNodeEvent(node)
+	}
+
+	var err error
+	if syncZoneIc && config.OVNKubernetesFeature.EnableInterconnect {
+		if err = oc.zoneICHandler.AddRemoteZoneNode(node); err != nil {
+			klog.Errorf("Failed to add the remote zone node [%s] to the zone interconnect handler, err : %v", node.Name, err)
```

Comment:
do you need to log error here? Why not just Store the failure and then return this as the error?

Reply:
Ack